pkgs/development/libraries/libspatialindex/default.nix

```diff
@@ -1,20 +1,26 @@
-{ lib, stdenv, fetchurl }:
+{ lib, stdenv, fetchFromGitHub, cmake }:
 
 stdenv.mkDerivation rec {
   pname = "libspatialindex";
-  version = "1.8.5";
+  version = "1.9.3";
 
-  src = fetchurl {
-    url = "https://download.osgeo.org/libspatialindex/spatialindex-src-${version}.tar.gz";
-    sha256 = "1vxzm7kczwnb6qdmc0hb00z8ykx11zk3sb68gc7rch4vrfi4dakw";
+  src = fetchFromGitHub {
+    owner = "libspatialindex";
+    repo = "libspatialindex";
+    rev = version;
+    hash = "sha256-zsvS0IkCXyuNLCQpccKdAsFKoq0l+y66ifXlTHLNTkc=";
   };
 
-  enableParallelBuilding = true;
+  nativeBuildInputs = [ cmake ];
 
-  meta = {
+  cmakeFlags = lib.optional doCheck "-DSIDX_BUILD_TESTS=ON";
```

Comment:
FWIW, if you used the `finalAttrs` mechanism to reference `doCheck`, it would work as expected when someone does a `.overrideAttrs (_: {doCheck = false;})`

Reply:
Never heard before about this way of declaring derivation. As I see it's not widely used in nixpkgs (38 derivations only). Is it somehow new/recommended way? Or in what cases it would be good to use?

Reply:
It's moderately new but also under-advertized. It is documented though: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes